### DISCUSSION TITLE
How to get the currently recognized plate ?

### DISCUSSION THREAD

**Discussion by GitKiller-HA** [robmarkcole/HASS-plate-recognizer#73] [Q&A] [answered]
Hi,

There is an example how to compare the currently read plate with the definition of "wached_plates", but how can I get the currently read plate as a sensor ?

I try something like below to get currently read plate and confidence of read:

>_current_plate_recognizer:
>value_template: "{{ state_attr('image_processing.platerecognizer_xxxcamera', 'vehicles').plate }}"_   

>_current_plate_recognizer_confidence:
>value_template: "{{ state_attr('image_processing.platerecognizer_xxxcamera', 'vehicles').confidence }}"_   

but I get warning:

> _Template variable warning: 'list object' has no attribute 'plate' when rendering '{{ state_attr('image_processing.platerecognizer_xxxcamera', 'vehicles').plate }}'_

>_Template variable warning: 'list object' has no attribute 'confidence' when rendering '{{ state_attr('image_processing.platerecognizer_xxxcamera', 'vehicles').confidence }}'_

Regards,
Grzegorz

**Comment by GitKiller-HA** [comment, marked as answer]
Any one ?

**Reply by mprusinowski** [reply]
Any solution? Same problem here

**Reply by HAJozefow** [reply]
Hi,

I wrote sensor like below and it generate warning, but works fine.

  plate_recognizer_plates:
    friendly_name: "License plates"
    icon_template: mdi:car
    value_template: >-
      {%- if (state_attr('image_processing.platerecognizer_camera', 'vehicles') |count) == 0 %}
        None
      {% else %}
        {%- for i in range(0, (state_attr('image_processing.platerecognizer_camera', 'vehicles') |count) )%}
        {{- state_attr('image_processing.platerecognizer_camera', 'vehicles')[i].plate | upper}}
        {% endfor %}
      {% endif %}

**Reply by HAJozefow** [reply]
If you don't want warnings you should get object vehicle. This return string which you should parsing to get all interesting information
  vehicle_recognizer:
    value_template: "{{ state_attr('image_processing.platerecognizer_gatecamera_fluent', 'vehicles') }}"
